algorithms/linfa-elasticnet/src/algorithm.rs

```diff
@@ -130,6 +216,62 @@ impl<F: Float> ElasticNet<F> {
     }
 }
 
+/// View the fitted parameters and make predictions with a fitted
+/// elastic net model
+impl<F: Float> MultiTaskElasticNet<F> {
+    /// Get the fitted hyperplane
+    pub fn hyperplane(&self) -> &Array2<F> {
+        &self.hyperplane
+    }
+
+    /// Get the fitted intercept, [0., ..., 0.] if no intercept was fitted
+    /// Note that there are as many intercepts as tasks
+    pub fn intercept(&self) -> &Array1<F> {
+        &self.intercept
+    }
+
+    /// Get the number of steps taken in optimization algorithm
+    pub fn n_steps(&self) -> u32 {
+        self.n_steps
+    }
+
+    /// Get the duality gap at the end of the optimization algorithm
+    pub fn duality_gap(&self) -> F {
+        self.duality_gap
+    }
+
+    /// Calculate the Z score
+    pub fn z_score(&self) -> Result<Array1<F>> {
+        self.variance
+            .as_ref()
+            .map(|variance| {
+                self.hyperplane
+                    .iter()
+                    .zip(variance.iter())
+                    .map(|(a, b)| *a / b.sqrt())
+                    .collect()
+            })
+            .map_err(|err| err.clone())
+    }
+
+    /// Calculate the confidence level
+    pub fn confidence_95th(&self) -> Result<Array1<(F, F)>> {
+        // the 95th percentile of our confidence level
+        let p = F::cast(1.645);
+
+        self.variance
+            .as_ref()
+            .map(|variance| {
+                self.hyperplane
+                    .iter()
+                    .zip(variance.iter())
+                    .map(|(a, b)| (*a - p * b.sqrt(), *a + p * b.sqrt()))
+                    .collect()
+            })
+            .map_err(|err| err.clone())
+    }
```

Comment:
These 2 methods are identical between the 2 `ElasticNet`s, so they should be refactored out.